src/libostree/ostree-bootloader-uboot.c

```diff
@@ -95,7 +95,45 @@ create_config_from_boot_loader_entries (OstreeBootloaderUboot     *self,
 
   val = ostree_bootconfig_parser_get (config, "options");
   if (val)
-    g_ptr_array_add (new_lines, g_strdup_printf ("bootargs=%s", val));
+    {
+      glnx_fd_close int uenv_fd = -1;
+      g_autofree char* kargs = g_strdup (val);
+      const char *uenv_path = NULL;
+      const char *ostree_arg = NULL;
+
+      g_ptr_array_add (new_lines, g_strdup_printf ("bootargs=%s", val));
+
+      /* Append system's uEnv.txt, if it exists in $deployment/usr/lib/ostree-boot/ */
+      ostree_arg = strtok (kargs, " ");
+      while (ostree_arg != NULL && !g_str_has_prefix (ostree_arg, "ostree="))
+        ostree_arg = strtok (NULL, " ");
+      if (!ostree_arg)
+      {
+        g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                             "No ostree= kernel argument found in boot loader configuration file");
+        return FALSE;
+      }
+      uenv_path = glnx_strjoina (ostree_arg + strlen ("ostree=/"), "/usr/lib/ostree-boot/uEnv.txt");
+      uenv_fd = openat (self->sysroot->sysroot_fd, uenv_path, O_CLOEXEC | O_RDONLY);
+      if (uenv_fd != -1)
+        {
+          char *uenv = glnx_fd_readall_utf8 (uenv_fd, NULL, cancellable, error);
+          if (!uenv)
+            {
+              glnx_set_prefix_error_from_errno (error, "%s", uenv_path);
```

Comment:
This needs to be  `g_prefix_error (error, "Reading %s: ", uenv_path);`.  Here we already have a `GError`, we need to prepend to it.